core/chains/evm/txmgr/evm_tx_store.go

```diff
@@ -1292,26 +1292,25 @@ func (o *evmTxStore) SaveInProgressAttempt(ctx context.Context, attempt *TxAttem
 	return nil
 }
 
-func (o *evmTxStore) GetNonFatalTransactions(ctx context.Context, chainID *big.Int) (txes []*Tx, err error) {
+func (o *evmTxStore) GetNonFatalTransactionsByBatch(ctx context.Context, chainID *big.Int, offset, limit uint) (txes []*Tx, err error) {
```

Comment:
I think the overflow was because the previous query was too generic, returning a huge amount of transactions.
I think you can filter the query to only return txs with status = "unconfirmed".
Also if you can filter for txs that have fromAddress not in the enabledAddresses list.

If you do the above, then you don't need to use batch or pagination, as the list of items returned will be small.

In future, once we implement the status Finalized, we can return txs with state unconfirmed and confirmed. But till we do that, lets not track txs that are in confirmed state.

Reply:
If we only query for `TxUnconfirmed` then we will miss marking `TxInProgress` and `TxUnstarted` txes as fatal. I think in this case it's better to query for everything except fatal and confirmed, and I'll add the enabledAddresses check!

Reply:
I think it's worth tracking and handling all tx states. Especially since we know that TxInprogress and TxUnstarted will never be confirmed if they were abandoned. We already have the finalization check in the tracker as well, so I don't see a reason to not track confirmed txes. Of course we can refactor the confirmed case to check for finalized status instead once that's implemented. I don't think it's worth reducing functionality to avoid this query since it only happens once on startup.

I'll add the check for enabledAddrs in the query. I think it may still be worth batching here just for safety however.